dataprofiler/profilers/histogram_utils.py

```diff
@@ -14,6 +14,293 @@
 )
 
 
+def _ptp(maximum, minimum):
+    """Peak-to-peak using minimum and maximum value of a dataset.
+
+    This implementation avoids the problem of signed integer arrays having a
+    peak-to-peak value that cannot be represented with the array's data type.
+    This function returns an unsigned value for signed integer arrays.
+    Function follows the numpy implementation within:
+    https://github.com/numpy/numpy/blob/main/numpy/lib/histograms.py
+    """
+    return np.subtract(maximum, minimum)
+
+
+def _calc_doane_bin_width_from_profile(profile):
+    """
+    Doane's histogram bin estimator reworked to use profiles.
+
+    Improved version of Sturges' formula which works better for
+    non-normal data. See
+    stats.stackexchange.com/questions/55134/doanes-formula-for-histogram-binning
+    Function follows the numpy implementation within:
+    https://github.com/numpy/numpy/blob/main/numpy/lib/histograms.py
+
+    Parameters
+    ----------
+    profile : NumericStatsMixin
+        Input data's data profile that is to be histogrammed
+
+    Returns
+    -------
+    An estimate of the optimal bin width for the given data.
+    """
+    try:
+        dataset_size = profile.match_count
+    except AttributeError:
+        dataset_size = sum(profile._stored_histogram["histogram"]["bin_counts"])
```

Comment:
We do this a lot. Maybe abstract to a function we use at the top instead?

Reply:
I think the correct way to do it would be to calculate it one time and use it like how I had it in the temporary rework I indicated previously. We can turn it into a function temporarily but I think the actual fix will come later when we make this code work together with the pre-existing code such that this number will only need to be calulated one time.

Reply:
Agreed, a better system. But, until then probably still ideal to nav towards dry. no extra tests needed.